### ISSUE TITLE
supabase-js 2.0rc4 - "This endpoint requires a Bearer token" calling signOut()

### ISSUE THREAD

**Issue by howesteve** [supabase/supabase-js#530]
Hello,

I'm already using version 2.0.0-rc.4. So in the code below, user is signed in (100% sure), and when I call ```signOut()```:

```js
const { error } = await supabase.auth.signOut();
console.log(error.message);
```
... the following error is thrown in the chrome debug console:
```
helpers.ts:40          GET https://xxxxxxxxxxxxxx.supabase.co/auth/v1/user 401
common.ts:91 AuthApiError: This endpoint requires a Bearer token
    at fetch.ts:36:16
```
Anyway, even after the error message, the session is successfully closed. This was working fine on the 1.x branch. What am I doing wrong?

Thanks

**Comment by alaister** [comment]
Hi @howesteve,
Can you confirm this is still an issue on [2.0.0-rc.8](https://www.npmjs.com/package/@supabase/supabase-js/v/2.0.0-rc.8)?

**Comment by howesteve** [comment]
Yes, it's still the same with [2.0.0-rc.8](https://www.npmjs.com/package/@supabase/supabase-js/v/2.0.0-rc.8).

**Comment by alaister** [comment]
Hey @fernandops26,

Are you calling `getUser()` when the user is not logged in? That would be the correct error message in that case.

`getUser()` is not a direct replacement for `user()` in V1, as it makes an HTTP request every time it's called for the user. Here's how I'd get the user in V2:
```typescript
const {
  data: { session },
  error,
} = await supabase.auth.getSession()

if (error) {
  // handle error
}

const user = session?.user
```

@howesteve you wouldn't happen to be calling `getUser()` somewhere in your code as well?

**Comment by howesteve** [comment]
> @howesteve you wouldn't happen to be calling `getUser()` somewhere in your code as well?
Yes, I was actually calling `getUser()` inside a ```onAuthStateChange()``` handler. Now that I switched into a ```getSession()``` -only approach, it looks fixed. However, that was actually quite confusing, since it's supposed to be regular api.

Thanks

**Comment by mandarini** [comment]
I will be closing this issue since it's 3 years old, and also it seems to have a solution. If you think that this issue still persists, let me know, and please open a new issue with minimal reproduction, to help me track it better! Thank you all for contributing!! :D
